website/prebuild.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+"""Updates files in preparation for a jekyll build.
+
+Used from .github/workflows/gh_pages.yaml. This updates the file and directory
+structure prior to the jekyll build.
+"""
+
+import dataclasses
+import os
+from pathlib import Path
+import re
+from typing import Optional
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+
+@dataclasses.dataclass
+class ChildDir:
+    """Tracks whether a child directory has grandchildren."""
+
+    title: str
+    has_grandchildren: bool = False
+
+
+def get_title(f: Path, content: str) -> str:
+    """Returns a file's title according to markdown.
+
+    Replacements are for YAML compatibility in `add_frontmatter`.
+    """
+    m = re.search("^# (.*)$", content, re.MULTILINE)
+    assert m, f
+    title = m[1]
+    title = title.replace("\\", "\\\\")
+    title = title.replace('"', '\\"')
```

Comment:
It'd be good to also convert `` ` ``s appearing in page titles into `<code>` ... `</code>`; we have a few of those in the design.

Reply:
Done